mindmeld/query_factory.py

```diff
@@ -30,16 +30,26 @@ class QueryFactory:
         tokenizer (Tokenizer): the object responsible for normalizing and tokenizing processed
             text
         stemmer (Stemmer): the object responsible for stemming the text
+        language (str): the language of the text
+        locale (str): the locale of the text
+        sys_recognizer (SystemEntityRecognizer): the system entity recognizer, default to Duckling
     """
 
     def __init__(
-        self, tokenizer, preprocessor=None, stemmer=None, locale=None, language=None
+        self,
+        tokenizer,
+        preprocessor=None,
+        stemmer=None,
+        locale=None,
+        language=None,
+        sys_recognizer=None,
     ):
         self.tokenizer = tokenizer
         self.preprocessor = preprocessor
         self.stemmer = stemmer
         self.locale = locale
         self.language = language
+        self.sys_recognizer = sys_recognizer or DucklingRecognizer.get_instance()
```

Comment:
Again, lets decide on exactly one place where `sys_recognizer = sys_recognizer or DucklingRecognizer.get_instance()` pattern is done and all other users should except `sys_recognizer` and not try to access the singleton.

Reply:
We can't because `QueryFactory` can be used independently from the application.